src/Raven.Server/Documents/DocumentsStorage.cs

```diff
@@ -510,6 +510,17 @@ public IEnumerable<Document> GetDocumentsFrom(DocumentsOperationContext context,
             }
         }
 
+        public IEnumerable<Document> GetDocumentsFrom(DocumentsOperationContext context, List<string> collections, long etag, int take)
+        {
+            foreach (var collection in collections)
+            {
+                foreach (var document in GetDocumentsFrom(context, collection, etag, 0, take))
```

Comment:
you need to handle 'take' manually. Consider a situation when you have take = 1024 and you have 10 collections. You do not want to export 10 x 1024 docs. Just 1024.